addons/rmsmartshape/shapes/shape.gd

```diff
@@ -919,6 +919,9 @@ func bake_collision() -> void:
 	else:
 		generated_points = _generate_collision_points_precise()
 
+	if generated_points.size() > 1 and generated_points[0] == generated_points[-1]:
+		generated_points = generated_points.slice(0, -1)
+
```

Comment:
Don't use slice(), it makes another needless copy.

1. Get points (`_generate_collision_points_*`)
2. Compute the transform matrix
3. Apply the transform. This returns a new array with transformed points.
4. Perform the duplication check
5. If needed, resize the array to size -1
6. Assign the array to the collision polygon

Also please squash your commits when you're done.